### DISCUSSION TITLE
Impact of React 19's Native Web Component Support on ui5-webcomponents-react Roadmap

### DISCUSSION THREAD

**Discussion by prsdthkr** [UI5/webcomponents-react#6717] [General]
With React 19 introducing enhanced native support for Custom Elements/Web Components, what are the implications for the long-term roadmap of ui5-webcomponents-react?

See https://react.dev/blog/2024/12/05/react-19#support-for-custom-elements.

**Comment by MarcusNotheis** [comment]
Hi @prsdthkr,

thanks for reaching out!
The support for Custom Elements in React 19 will not affect our roadmap as of now, because there are still some benefits in using our UI5 Web Components for React wrapper over UI5 Web Components directly in React:

1. Automatic typeahead completions when using the components. If you would use the `ui5-button` component directly, you would have to register this tag in the JSX namespace first.
2. Custom Components like charts, ObjectPage, AnalyticalTable: these components are still build on top of UI5 Web Components
3. If you use features like custom element scoping, you don't need to change all tag names in your app as the React wrappers will update their internal tag name automatically.

In addition, you still benefit from the React 19 Custom Element support as we have an internal feature detection for React 19 which will bind the properties and events in the new recommended way.
